liquid-rust-typeck/src/binding_tree.rs

```diff
@@ -1,30 +1,33 @@
-use std::fmt;
+use std::{collections::BTreeMap, fmt};
 
-use liquid_rust_common::{index::IndexMap, new_index, ordered_hash_map::OrderedHashMap};
+use liquid_rust_common::index::{newtype_index, IndexVec};
 use liquid_rust_fixpoint::{Constraint, Fixpoint};
-use liquid_rust_lrir::ty::{BaseTy, Pred, Refine, Ty, Var};
+use liquid_rust_lrir::ty::{Refine, Ty, Var};
 
-new_index! {
-    NodeId
+newtype_index! {
+    struct NodeId {
+        DEBUG_FORMAT = "node{}",
+        const FIRST_NODE = 0
+    }
 }
 
 pub struct BindingTree {
-    nodes: IndexMap<NodeId, Node>,
+    nodes: IndexVec<NodeId, Node>,
     curr_path: Vec<NodeId>,
-    curr_bindings: OrderedHashMap<Var, Ty>,
+    curr_bindings: BTreeMap<Var, Ty>,
```

Comment:
Thinking over this, now I believe we can even use hash map directly. Originally, I was using an "ordered hash map" to preserve the order of binding as they were inserted, but since we keep track of that in `curr_path` I think it's not necessary to do anything fancy anymore.